streams/src/test/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilderTest.java

```diff
@@ -156,6 +156,30 @@ public void shouldStillMaterializeSourceKTableIfMaterializedIsntQueryable() thro
         assertEquals("topic2", topology.storeToChangelogTopic().get(storeName));
         assertNull(table1.queryableStoreName());
     }
+    
+    @Test
+    public void shouldBuildGlobalTableWithNonQueryableStoreName() throws Exception {
+        final GlobalKTable<String, String> table1 = builder.globalTable("topic2",
+                                      consumed,
+                                      new MaterializedInternal<>(
+                                              Materialized.<String, String, KeyValueStore<Bytes, byte[]>>with(null, null),
+                                              builder,
+                                              storePrefix));
+
+        assertNull(table1.queryableStoreName());
+    }
+
+    @Test
+    public void shouldBuildGlobalTableWithQueryableStoreName() throws Exception {
+        final GlobalKTable<String, String> table1 = builder.globalTable("topic2",
+                                      consumed,
+                                      new MaterializedInternal<>(
```

Comment:
You can pass in a `Materialized` here directly -- no need to wrap it with `MaterializedInternal` (that should not be use anyway, as it's an internal class as it's name and package indicates).

Reply:
The ```globalTable``` method signature is as follows:
```public <K, V> GlobalKTable<K, V> globalTable(final String topic,
                                                 final ConsumedInternal<K, V> consumed,
                                                 final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>>                 materialized)``` 
Note that the method requires ```MaterializedInternal```, so ```Materialized``` cannot be used directly. A cast must be provided.